### PR TITLE
Partest: use a `// java: -flags` comment instead of .javaopts file

### PR DESCRIPTION
**Pull request by lrytz**
as promised here: https://github.com/scala/scala/pull/9587/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R133

### CODE REVIEW

File: src/partest/scala/tools/partest/package.scala [hunk 1]
```diff
@@ -143,18 +141,6 @@ package object partest {
 
   def basename(name: String): String = Path(name).stripExtension
 
-  /** In order to allow for spaces in flags/options, this
-   *  parses .flags, .javaopts, javacopts etc files as follows:
-   *  If it is exactly one line, it is split (naively) on spaces.
-   *  If it contains more than one line, each line is its own
-   *  token, spaces and all.
```
Comment (som-snytt) [inline src/partest/scala/tools/partest/package.scala:150]: I don't remember which test needed that, but IIRC toolArgs does quoted tokens.

File: src/partest/scala/tools/partest/nest/Runner.scala [hunk 2]
```diff
@@ -130,25 +130,15 @@ class Runner(val testInfo: TestInfo, val suiteRunner: AbstractRunner) {
...
-    //
-    // debug: Found javaopts file 'files/shootout/message.scala-2.javaopts', using options: '-Xss32k'
-    // debug: java -Xss32k -Xss2m -Xms256M -Xmx1024M -classpath [...]
+      suiteRunner.verbose(s"Using java options: '${javaopts.mkString(",")}'")
+
     val propertyOpts = propertyOptions(fork = true).map { case (k, v) => s"-D$k=$v" }
 
     val classpath = joinPaths(extraClasspath ++ testClassPath)
 
+    // `javaopts` last; for repeated arguments, the last one wins
```
Comment (som-snytt) [inline src/partest/scala/tools/partest/nest/Runner.scala:141]: very cool-headed and succinct, compared to the outrage of the previous comment.

**Review by som-snytt — APPROVED** [review]
toolArgs on a roll.

**Comment by lrytz** [comment]
Very funny test failure:
```
Could not find or load main class 115)]
```

Due to:

https://github.com/scala/scala/blob/d112b8c9b0274f5839875fd5a6c75c71fa81dc80/test/files/run/t4415.scala#L6

**Comment by som-snytt** [comment]
Ha! I was actually thinking that colon in Scala 3 would result in some false detections.

I think toolArgs takes about ten lines at the top? It's similar to when people do `catch ex: Exception` and screw up my vim modeline.

**Pull request merged**